### ISSUE TITLE
Updates to Issues Possible Today?

### ISSUE THREAD

**Issue by brianjking** [dear-github/dear-github#125]
https://github.com/owncloud/core/issues/new

<img width="895" alt="screenshot2-17-1609 56" src="https://cloud.githubusercontent.com/assets/6786817/13115394/3700bde6-d55d-11e5-8b19-d20f67ff0eb0.png">

**Comment by aensley** [comment]
I was wondering what new update you were pointing out until I clicked on the link. The template is preloaded. Nice! I certainly hope this is coming to everyone soon!

**Comment by aensley** [comment]
Looks like it's coming from this file: https://github.com/owncloud/core/blob/master/issue_template.md

**Comment by aensley** [comment]
The link to the contributing file has been there as long as I've been using GH. The pre-loaded template is new to me though. Testing it out on my repo right now.

**Comment by aensley** [comment]
It works on my repo! Awesome!

**Comment by brianjking** [comment]
@aensley Did you just create a issue_template.md file at the root of your repo and it automatically inserts this?

**Comment by aensley** [comment]
Yes. Exactly. I'd give you a link, but it's a private repo.

**Comment by brianjking** [comment]
@aensley I can confirm it's working for me also! Cool! 

Yeah, totally gapped about the contributing link always being there.

**Comment by looterz** [comment]
This is awesome!

**Comment by joshmanders** [comment]
Great news!

**Comment by tikotzky** [comment]
The issue template in that repo was originally created in Jan 2013. 

Did this functionality exist the whole time?

**Comment by joshmanders** [comment]
Some plugins and that exist.

One thing I noticed is it uses `issue_template.md` for the source of it. I'd much rather have a setting page on the repo to set the markdown for it manually, I don't want a bunch of github specific files mucking up my repos.

**Comment by kentcdodds** [comment]
> I'd much rather have a setting page on the repo to set the markdown for it manually, I don't want a bunch of github specific files mucking up my repos.

Agreed, but this is nice to have.

**Comment by joshmanders** [comment]
Indeed it is. I'll welcome it with open arms.

**Comment by josegonzalez** [comment]
Doesn't affect mobile web maybe?

**Comment by bfontaine** [comment]
> I'd much rather have a setting page on the repo to set the markdown for it manually, I don't want a bunch of github specific files mucking up my repos.

The issue with that is you couldn’t make PR to fix issues in the issue template itself.

**Comment by brianjking** [comment]
@joshmanders Yeah, I agree. However, this is a great addition (or potentially there the whole time without any documentation).

I'm not sure how it will impact things like Sphinx-Doc or Jekyll sites either having this random Markdown document in the root of the repo.

**Comment by brianjking** [comment]
@josegonzalez It doesn't work with Chrome on my Nexus 6P or ForkHub on Android. :frowning: 

Either way, it's a start. :+1:

**Comment by joshmanders** [comment]
@bfontaine You have a valid point there. @brianjking the repo shown in this issue has had the `issue_template.md` file since 2013. I assume it's part of a browser plugin or something that pre-fills in the issue with the contents for you if you have the plugin installed.

**Comment by Rican7** [comment]
@brianjking I'm so curious... how did you find out about this? Just went to create an issue on a repo and found the feature by accident?

**Comment by svenluijten** [comment]
@joshmanders I don't have any extension installed, tested in incognito as well. It seems to work for me.

**Comment by KellerFuchs** [comment]
@joshmanders It's not a browser plugin: works for me on Tor Browser 5.5alpha with JS enabled (and no extra plugins)

**Comment by josegonzalez** [comment]
@brianjking for sure :)

**Comment by joshmanders** [comment]
@svenluijten & @KellerFuchs I never said this was a plugin, I was stating the `issues_template.md` from the `owncloud/core` repo is probably an artifact from such a plugin used by a member of the team there.

**Comment by josegonzalez** [comment]
Anyone know if it can be used with any casing of `issue_template.md`? I think we've normally used stuff like `README.md` and `CONTRIBUTING.md`, so I wonder if that extends to this.

**Comment by KellerFuchs** [comment]
@josegonzalez Try it, I guess.

**Comment by KellerFuchs** [comment]
@joshmanders  Aaaah, ok.

**Comment by svenluijten** [comment]
@joshmanders Ah, gotcha.

**Comment by josegonzalez** [comment]
WORKS: https://github.com/dokku/dokku/issues/new

**Comment by jwalton** [comment]
Is there a PR equivalent, I wonder?

**Comment by colinodell** [comment]
@jwalton Yep, `pull_request_template.md` seems to work: https://github.com/colinodell/testing/blob/master/pull_request_template.md

**Comment by brianjking** [comment]
@Rican7 Randomly via Hacker News.

**Comment by brianjking** [comment]
If it's not a plugin then it must be magic, right? 

![giphy 1](https://cloud.githubusercontent.com/assets/6786817/13119330/33f47514-d56e-11e5-9aba-234832cfcf90.gif)

**Comment by jwalton** [comment]
@colinodell Nice!  Thanks!

**Comment by brianjking** [comment]
If this turns out to be something that has been there the whole time without any documentation written about it... :facepunch:

**Comment by jwalton** [comment]
@brianjking Wat?  All this time my team has been copy-pasting templates from our wiki.  -_-

**Comment by kevinSuttle** [comment]
You'd think these 2 would just be headings in Contributing.md...

**Comment by meredevelopment** [comment]
@brianjking aye, doing a search for that file github-wide does show it in some pretty old repos 🙈

**Comment by calvinf** [comment]
@josegonzalez I'm not seeing it on mobile yet either.

**Comment by iwaffles** [comment]
Very cool stuff, thanks for sharing! Going to give this a shot!

**Comment by nobso** [comment]
This works! Thanks.

**Comment by orliesaurus** [comment]
This is really sick - thanks for listening!

**Comment by PVince81** [comment]
> If this turns out to be something that has been there the whole time without any documentation written about it...

ownCloud had that file since a very long time ago. We just discovered today when clicking on "New issue" that Github started using it. Could be luck that it was named that way already :smile:

**Comment by aensley** [comment]
My guess is that "GitHub is listening," saw this in the wild via a third-party, and integrated it natively using the same naming convention.

**Comment by captn3m0** [comment]
The owncloud issue_template.md file has been there for 3 years: [#1160](https://github.com/owncloud/core/pull/1160) is the PR that added it and @LukasReschke doesn't mention any existing plugin there. Just a lucky fluke, I guess.

**Comment by joshmanders** [comment]
@captn3m0 Interesting, was this a previous feature of github in 2013 and they removed it?

**Comment by LukasReschke** [comment]
> The owncloud issue_template.md file has been there for 3 years: #1160 is the PR that added it and @LukasReschke doesn't mention any existing plugin there. Just a lucky fluke, I guess.

This is correct. As far I am aware there is no plugin support for this and we have just used to link to this from our contributing.md file. (which was actually nearly always ignored :see_no_evil:)

As @PVince81 in https://github.com/dear-github/dear-github/issues/125#issuecomment-185344126 stated we discovered this today randomly while filing an issue. (see also [HN](https://news.ycombinator.com/item?id=11116792))

> @captn3m0 Interesting, was this a previous feature of github in 2013 and they removed it?

No.

**Comment by joshmanders** [comment]
@LukasReschke sorry your HN article didn't get any traction. :(

**Comment by LukasReschke** [comment]
> @LukasReschke sorry your HN article didn't get any traction. :(

There's worse than that :wink: Happy to have this feature at least after having asked GitHub support like 1000 times for it :smile:

**Comment by svenluijten** [comment]
Guess it's official now :wink: https://github.com/blog/2111-issue-and-pull-request-templates

**Comment by fommil** [comment]
this is really fantastic. It'd be even better if new tickets opened in Preview mode so our lovely formatting was preserved :smile:
